docs/src/topics/reference_shapes.md

```diff
@@ -0,0 +1,85 @@
+# Reference shapes
+
+The reference shapes in Ferrite are used to define grid cells,
+function interpolations (i.e. shape functions), and quadrature rules.
+Currently, the following reference shapes are defined
+
+* `RefLine`
+* `RefTriangle`
+* `RefQuadrilateral`
+* `RefTetrahedron`
+* `RefHexahedron`
+* `RefPrism`
+* `RefPyramid`
+
+## Entity naming
+Ferrite denotes the entities of a reference shape as follows
+
+| Entity   | Description |
+| :------- | :---------- |
+| `Vertex` | 0-dimensional points in the reference shape. |
+| `Edge`   | 1-dimensional entities connecting two vertices. |
+| `Face`   | 2-dimensional entities whose boundary is defined by 3 or more edges. |
+| `Volume` | 3-dimensional entity enclosed by faces. |
+
+Note that a `Node` in Ferrite is not the same as a `Vertex`.
+Vertices denote endpoints of edges, while nodes may also be be located in the middle
+of edges (e.g. for a `QuadraticLine` cell).
```

Comment:
```suggestion
Note that a `Node` in Ferrite is not the same as a `Vertex`.
Vertices denote endpoints of lines and the corners shapes with dimension larger than one, while nodes may also be located in the interior of edges, faces and volumes, acting as control points for the (geometric) non-linearity. Think for example about a `QuadraticLine` cell which has one node at the center of an edge and two more nodes located at the vertices/endpoints.
```

I think we can expand on this a bit, since this is really important to understand for users. I would suggest something in this direction, but it needs further refinement I think.

Reply:
I thought that for interpolations we considered the middle point as a vertex and that node was just "points in a grid"...

Reply:
It is a point in the grid because we enforce nodal interpolations for the geometry. :P

Reply:
I think of (maybe influenced by @termi-official) that we have two types of nodes
* Geometric nodes: Points in the grid (technically support points for the geometric interpolation).
* Algebraic nodes: Support points for nodal interpolations (e.g. Lagrange interpolations, but not Nedelec), as referred to [here](https://finite-element.github.io/L2_fespaces.html#ciarlet-s-finite-element). The coordinates of these nodes are not explicitly defined in the grid, but obtained by mapping their location in the reference point by the geometric interpolation of the cell. 

I've thought that it would make sense to put this in a separate doc. I don't want to put the emphasis here, since that is too hidden and takes away focus from the main point (dealing with reference shapes, and nodes haven't been introduced yet). 

I'll open a separate issue where we could list which terms could be defined on a "defintions" page, which would be good to have IMO.

Reply:
> I think of (maybe influenced by @termi-official) that we have two types of nodes
> 
>     * Geometric nodes: Points in the grid (technically support points for the geometric interpolation).
> 
>     * Algebraic nodes: Support points for nodal interpolations (e.g. Lagrange interpolations, but not Nedelec), as referred to [here](https://finite-element.github.io/L2_fespaces.html#ciarlet-s-finite-element). The coordinates of these nodes are not explicitly defined in the grid, but obtained by mapping their location in the reference point by the geometric interpolation of the cell.
> 

*achievement unlocked* :sunglasses: 

> 
> I've thought that it would make sense to put this in a separate doc. I don't want to put the emphasis here, since that is too hidden and takes away focus from the main point (dealing with reference shapes, and nodes haven't been introduced yet).
> 
> I'll open a separate issue where we could list which terms could be defined on a "defintions" page, which would be good to have IMO.

Makes sense. Should we then just revert the change here and add to the issue that we might want to put a reference in this page to the explanation for the nodes?

Reply:
I think the short note here is fine for now, and once we have a good list we can shorten it more or even remove it completely and rather link to the definitions page. 

> I'll open a separate issue

#944